From: Jiri Slaby <jslaby@suse.cz>
Subject: ORC: fix retpolines segfaults
Patch-mainline: never, SUSE specific
References: bnc#1068032 CVE-2017-5715

With the non-upstream patch:
 patches.suse/0013-x86-entry-Stuff-RSB-for-entry-to-kernel-for-non-SMEP.patch
ORC unwinder crashes due to the functions calling themselves and
insn->call_dest is NULL that case. So handle this case, meaning ORC info
is not valid here.

When the above patch is in upstream, this patch is to be removed too.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 tools/objtool/check.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/tools/objtool/check.c
+++ b/tools/objtool/check.c
@@ -1744,6 +1744,9 @@ static int validate_branch(struct objtoo
 			return 0;
 
 		case INSN_CALL:
+			if (!insn->call_dest)
+				break;
+
 			if (is_fentry_call(insn))
 				break;
 
